akka-http/src/main/scala/akka/http/javadsl/server/directives/SecurityDirectives.scala

```diff
@@ -52,8 +52,11 @@ abstract class SecurityDirectives extends SchemeDirectives {
    * Extracts the potentially present [[HttpCredentials]] provided with the request's [[akka.http.javadsl.model.headers.Authorization]] header.
    */
   def extractCredentials(inner: JFunction[Optional[HttpCredentials], Route]): Route = RouteAdapter {
-    D.extractCredentials { cred =>
-      inner.apply(cred.map(_.asJava).asJava).delegate // TODO attempt to not need map()
+    D.extractCredentials { credOpt =>
+      (credOpt match {
+        case Some(cred) => inner.apply(Optional.of(cred.asJava))
```

Comment:
Isn't that the same just written more verbosely? The TODO here basically was about whether `cred.asJava` is necessary at all. If the answer is yes, then the code can stay as is.

Reply:
returned as it was. I can't write `cred.asJava` because of collision with OptionConverters's asJava method, but if I remove this import, everything else breaks